### PR TITLE
Use Android NDK to r19.

### PR DESCRIPTION
**Pull request by juliusmik**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ endif
 JAVAFLAGS	= -XX:MaxPermSize=128M
 JAVACFLAGS  = -source 1.8 -target 1.8
 
-CPPFLAGS	+= -g0 -O2 -Wall -Werror -Wno-long-long -std=c++11
+CPPFLAGS	+= -g0 -Os -Wall -Werror -Wno-long-long -std=c++11 -fno-addrsig
```
Comment (Over17) [inline Makefile]: Why did you change the optimization level?
cc @ignas2

**Review by Over17 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -19,7 +19,7 @@ endif
 JAVAFLAGS	= -XX:MaxPermSize=128M
 JAVACFLAGS  = -source 1.8 -target 1.8
 
-CPPFLAGS	+= -g0 -Os -Wall -Werror -Wno-long-long -std=c++11 -fno-addrsig
+CPPFLAGS	+= -g0 -O2 -Wall -Werror -Wno-long-long -std=c++11 -fno-addrsig
```
Comment (ignas2) [inline Makefile:23]: @juliusmik Can you add a comment why performance optimization is needed here? So that in the future this flag is not toggled back and forth. If -Os produces smaller code size that's still good to know.
